src/nnet3bin/nnet3-split-egs.cc

```diff
@@ -0,0 +1,107 @@
+// nnet3bin/nnet3-split-egs.cc
+
+// Copyright 2012-2015  Johns Hopkins University (author:  Daniel Povey)
+//                2015  Yiming Wang
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include <sstream>
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "nnet3/nnet-example.h"
+#include "nnet3/nnet-example-utils.h"
+#include "nnet3/nnet-utils.h"
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+    using namespace kaldi::nnet3;
+    typedef kaldi::int32 int32;
+    typedef kaldi::int64 int64;
+
+    const char *usage =
+        "Split large chunks in examples into smaller chunks of equal size\n"
+        "by generating more examples containing these smaller chunks,\n"
+        "while still preserving the number of left/right context for\n"
+        "each new chunk.\n"
+        "\n"
+        "Usage:  nnet3-split-egs [options] <raw-model-in> <egs-in> <egs-out>\n"
+        "\n"
+        "An example:\n"
+        "nnet3-split-egs --chunk-size-after-split=20 1.raw \\\n"
+        "'ark:nnet3-merge-egs ark:1.egs ark:-|' ark:- \n";
+
+    int32 chunk_size_after_split = 0;
+        
+    ParseOptions po(usage);
+    po.Register("chunk-size-after-split", &chunk_size_after_split,
+		"used for enabling state preserving LSTM. default 0 indicates "
```

Comment:
State preserving training is applicable for any RNN, not just LSTMs. Rewrite your comments to reflect that. I remember seeing this at other places in the code.